aoc/src/main/scala/aoc2015/Day03.scala

```diff
@@ -0,0 +1,60 @@
+package aoc2015
+
+import scala.io.Source
+import scala.collection.mutable.ListBuffer
+
+object Day03 extends App:
+  val lines: String =
+    Source
+      .fromFile("aoc/src/resources/aoc2015/Day03")
+      .getLines
+      .mkString
+
+  def generateDirections(): Int =
+    var evenDirections: String = ""
+    var unevenDirections: String = ""
+    lines.zipWithIndex.foreach {
+      case (x, i) =>
+        if (i % 2 == 0)
+          evenDirections = evenDirections.concat(x.toString)
+        else
+          unevenDirections = unevenDirections.concat(x.toString)
+    }
+    val even: List[(Int,Int)] = location(0,0,evenDirections)
+    val oneven: List[(Int,Int)] = location(0,0,unevenDirections)
+    val coo: List[(Int,Int)] = even ++ oneven
+    coo.distinct.size
+
+
+  def location(currentx: Int, currenty: Int, directions: String, xAxis: List[Int] = List(0), yAxis: List[Int] = List(0)): List[(Int,Int)] =
+    if (directions.isEmpty) {
```

Comment:
perhaps implement the head direction part of the if/else tree with pattern matching on `Char`s, ie. between single-quotes

```
directions.head match
  case '^' =>
    val newx = currentx + 0
    val newy = currenty + 1
    location(newx, newy, directions.tail, xAxis :+ newx, yAxis :+ newy)
  case 'v' =>
    ...
```